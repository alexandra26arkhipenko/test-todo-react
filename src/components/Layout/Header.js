import classes from "./Header.module.css";
import todoImage from "../../assets/todo.jpg";

const Header = () => {
  return (
    <div className={classes['todo-image']}>
      <img src={todoImage} />
    </div>
  );
};

export default Header;
