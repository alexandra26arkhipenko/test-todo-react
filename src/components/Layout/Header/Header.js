import React from "react";
import classes from "./Header.module.css";
import todoImage from "../../../assets/todo.jpg";

const Header = () => {
  return (
    <header className={classes["todo-image"]}>
      <img src={todoImage} alt="plants" />
    </header>
  );
};

export default Header;
