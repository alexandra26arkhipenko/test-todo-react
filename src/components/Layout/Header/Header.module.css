.todo-image {
  width: 100%;
  height: 12rem;
  z-index: 0;
  overflow: hidden;
}

.todo-image img {
  width: 100%;
  height: 90%;
  object-fit: cover;
}
