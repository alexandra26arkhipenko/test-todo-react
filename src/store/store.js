import { configureStore } from "@reduxjs/toolkit";
import { localStorageMiddleware } from "./middleware/localStorageMiddleware";

import todoReducer from "./todo";

const store = configureStore({
  reducer: todoReducer,
  middleware: [localStorageMiddleware],
});

export default store;
