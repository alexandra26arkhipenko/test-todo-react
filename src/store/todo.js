import { createSlice } from "@reduxjs/toolkit";
import { v4 as uuid } from "uuid";
import { getItemFromLocalStorage } from "../repository/localStorageRepository";
import { TODO_FOOTER_FILTER_ALL } from "../constants/constants";

const savedTodos = getItemFromLocalStorage("todos");
let todos = [];

if (savedTodos) {
  todos = JSON.parse(savedTodos);
}

const initialState = {
  todos: todos,
  filter: "",
};

const todoSlice = createSlice({
  name: "counter",
  initialState,
  reducers: {
    addTodo(state, action) {
      const newTodo = {
        id: uuid(),
        text: action.payload,
        isCompleted: false,
      };

      state.todos.unshift(newTodo);
    },
    compliteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos[index].isCompleted = !state.todos[index].isCompleted;
      }
    },
    deleteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos.splice(index, 1);
      }
    },
    deleteAllCompletedTodos(state) {
      state.todos = state.todos.filter((todo) => !todo.isCompleted);
    },
    filterTodos(state, action) {
      state.filter = action.payload;
    },
    reorderTodos(state, action) {
      let { dragIndex, dropIndex, filteredTodos } = action.payload;

      if (state.filterValue !== TODO_FOOTER_FILTER_ALL) {
        dragIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dragIndex].id
        );
        dropIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dropIndex].id
        );
      }

      const updatedTodos = [...state.todos];
      const [reorderedItem] = updatedTodos.splice(dragIndex, 1);
      updatedTodos.splice(dropIndex, 0, reorderedItem);
      state.todos = updatedTodos;
    },
  },
});

export const todoActions = todoSlice.actions;
export default todoSlice.reducer;
