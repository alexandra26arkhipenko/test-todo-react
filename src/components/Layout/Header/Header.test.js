import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../store/store";
import Header from "./Header";

describe("Header", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
