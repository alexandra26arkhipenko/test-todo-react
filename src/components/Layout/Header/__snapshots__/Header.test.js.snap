// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <header
    class="todo-image"
  >
    <img
      alt="plants"
      src="todo.jpg"
    />
  </header>
</div>
`;
